flagengine/engine_eval/mappers.go

```diff
@@ -0,0 +1,383 @@
+				featureValue: featureValue,
+			})
+		}
+
+		// Generate hash for this set of overrides
+		overridesHash := generateHash(overrides)
+
+		// Group identifiers by their overrides
+		featuresToIdentifiers[overridesHash] = append(featuresToIdentifiers[overridesHash], identityOverride.Identifier)
+		overridesKeyToList[overridesHash] = overrides
+	}
+
+	// Create segment contexts for each unique set of overrides
+	segmentContexts := make(map[string]SegmentContext)
+
+	for overridesHash, identifiers := range featuresToIdentifiers {
+		overrides := overridesKeyToList[overridesHash]
+
+		// Create segment context
+		sc := SegmentContext{
+			Key:  "", // Identity override segments never use % Split operator
+			Name: "identity_overrides",
+			Rules: []SegmentRule{
+				{
+					Type: All,
+					Conditions: []Condition{
+						{
+							Operator: "IN",
+							Property: "$.identity.identifier",
+							Value:    &ValueUnion{String: func() *string { s := strings.Join(identifiers, ","); return &s }()},
+						},
+					},
+				},
+			},
+		}
+
+		// Create overrides for each feature
+		for _, override := range overrides {
+			priority := math.Inf(-1) // Strongest possible priority
+			featureOverride := FeatureContext{
+				Key:        "", // Identity overrides never carry multivariate options
+				FeatureKey: override.featureKey,
+				Name:       override.featureName,
+				Enabled:    override.enabled,
+				Priority:   &priority,
+			}
+
+			// Set the value if provided
+			if override.featureValue != "" {
+				featureOverride.Value = override.featureValue
+			}
+
+			sc.Overrides = append(sc.Overrides, featureOverride)
+		}
+
+		segmentContexts[overridesHash] = sc
+	}
+
+	return segmentContexts
+}
+
+// Trait represents a trait with key-value pair, compatible with the main package Trait struct.
+type Trait struct {
+	TraitKey   string      `json:"trait_key"`
+	TraitValue interface{} `json:"trait_value"`
+	Transient  bool        `json:"transient,omitempty"`
+}
+
+// MapContextAndIdentityDataToContext maps context and identity data to create an evaluation context
+// with identity information. This function takes an existing context and enriches it with identity
+// data including identifier and traits.
+func MapContextAndIdentityDataToContext(
+	context EngineEvaluationContext,
+	identifier string,
+	traits interface{},
+) EngineEvaluationContext {
+	// Convert traits to local type
+	var traitList []*Trait
+
+	if traits != nil {
+		// Handle different trait types by copying field values
+		switch v := traits.(type) {
+		case []*Trait:
+			traitList = v
+		default:
+			// Try to extract traits using reflection-like approach
+			// Since both Trait structs have the same JSON tags, we can marshal/unmarshal
+			if jsonBytes, err := json.Marshal(traits); err == nil {
+				if err := json.Unmarshal(jsonBytes, &traitList); err != nil {
+					// Log error or handle gracefully - for now, continue with empty list
+					traitList = nil
```

Comment:
could it work to handle them explicitely to avoid marshal/unmarshal?
Something like 
```
traitList = make([]*Trait, len(v))
for ... range {
  traitList[i] = &Trait {}
}
```
I think it should work to convert types but my go is a bit rusty